bot/exts/halloween/spooknamerate.py

```diff
@@ -0,0 +1,377 @@
+# Importing everything
+import asyncio
+import json
+import logging
+import random
+from calendar import month_name
+from collections import defaultdict
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Dict, Union
+
+from async_rediscache import RedisCache
+from discord import Embed, Message, Reaction, TextChannel, User
+from discord.ext import tasks
+from discord.ext.commands import Bot, Cog, Context, group
+
+from bot.constants import Channels, Client, Colours, Month
+from bot.utils.decorators import InMonthCheckFailure
+
+logger = logging.getLogger(__name__)
+MONTH_TO_LIMIT_TO = Month.OCTOBER
+
+EMOJIS_VAL = {  # the value of the emojis
+    '\N{Jack-O-Lantern}': 1,
+    '\N{Ghost}': 2,
+    '\N{Skull and Crossbones}': 3,
+    '\N{Zombie}': 4,
+    '\N{Face Screaming In Fear}': 5
+}
+
+
+class SpookNameRate(Cog):
+    """
+    A game that asks the user to spookify or halloweenify a name that is given everyday.
+
+    It sends a random name everyday. The user needs to try and spookify it to his best ability and
+    send that name back using the `spooknamerate add entry` command
+    """
+
+    messages = RedisCache()  # a variable to store all the messages
+    data = RedisCache()
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+
+        names_data = self.load_json(Path('bot', 'resources', 'halloween', 'spooknamerate_names.json'))
+        self.first_names = names_data['first_names']
+        self.last_names = names_data['last_names']
+        # the names are from https://www.mockaroo.com/
+
+        self.name = None
+
+        self.added_messages = ["Let's see if you win?", ":jack_o_lantern: SPOOKY :jack_o_lantern:",
+                               "If you got it, haunt it.", "TIME TO GET YOUR SPOOK ON! :skull:"]
+
+        self.bot.loop.create_task(self.load_vars())
+
+        self.first_time = None
+        self.poll = False
+        self.channel_id = Channels.community_bot_commands
```

Comment:
I'm not sure do we need to store it here, this is thing that should be hard-coded.

Reply:
You mean directly use `Channels.community_bot_commands`?

Reply:
Yes, there is no need for wrapper for this value.